click_odoo_contrib/copydb.py

```diff
@@ -26,11 +27,34 @@ def _copy_db(cr, source, dest):
     )
 
 
-def _copy_filestore(source, dest):
+def _copy_filestore(source, dest, copy_mode="default"):
     filestore_source = odoo.tools.config.filestore(source)
     if os.path.isdir(filestore_source):
         filestore_dest = odoo.tools.config.filestore(dest)
-        shutil.copytree(filestore_source, filestore_dest)
+        if copy_mode == "hardlink" or copy_mode == "rsync":
+            try:
+                if copy_mode == "hardlink":
+                    hardlink_option = ["--link-dest=" + filestore_source]
+                else:
+                    hardlink_option = []
+                cmd = (
+                    [
+                        "rsync",
+                        "-a",
+                        "--delete-delay",
+                    ]
+                    + hardlink_option
+                    + [
+                        filestore_source + "/",
+                        filestore_dest,
+                    ]
+                )
+                subprocess.check_call(cmd)
+            except Exception as e:
```

Comment:
nit:

```suggestion
            except subprocess.CalledProcessError as e:
```

Reply:
what does "nit:" mean?

Reply:
While the documentation for `.check_call` states it throws subprocess.CalledProcessError when the command does not succeed (return code != 0). It does not state what happens in other cases ;-)

Therefore i decided to catch all Exceptions here and wrap them in a ClickException with an additional hopefully helpfull message for the user invoking the script.

E.g.:
```bash
 python -c 'import subprocess; subprocess.check_call(["not_existing_command"])'
```
raises a FileNotFoundError:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python3.6/subprocess.py", line 306, in check_call
    retcode = call(*popenargs, **kwargs)
  File "/usr/lib/python3.6/subprocess.py", line 287, in call
    with Popen(*popenargs, **kwargs) as p:
  File "/usr/lib/python3.6/subprocess.py", line 729, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.6/subprocess.py", line 1364, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'not_existing_command': 'not_existing_command'
```

```bash
echo "not_executable" > not_executable
```
```
```

Reply:
> what does "nit:" mean?

It's an abbreviation for "nitpicking", i.e. a detail that I think (thought :) would improve the code but that is not very important in my eyes blocking. Thanks for the detailed explanation on why you chose `catch Exception`.